internal/provider/cached_image_resource.go

```diff
@@ -467,6 +466,8 @@ func runCacheProbe(ctx context.Context, builderImage string, opts eboptions.Opti
 	if err := os.MkdirAll(tmpKanikoDir, 0o755); err != nil {
 		return nil, fmt.Errorf("failed to create kaniko dir: %w", err)
 	}
+	// Use the temporary directory as our 'magic dir'.
+	opts.MagicDirBase = tmpKanikoDir
```

Comment:
(envbuilder nit: "base" makes me think of the directory that should contain the magicdir, i.e. `[base]/.envbuilder`).